### PR TITLE
[docs] Fix formatting issues and other guidelines in CNG doc

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

There are several formatting issues, few typos, writing styleguidelines not followed for consistency and the image isn't using `ImageSpotlight` component.

# How

By fixing the above mentioned issues.

# Test Plan

Changes have been tested locally.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/workflow/continuous-native-generation.mdx [hunk 1]
```diff
@@ -4,90 +4,91 @@ description: Learn about Continuous Native Generation (CNG) and how it works.
...
 <a
   aria-label="Expo Config Plugins downloads"
   href="http://www.npmtrends.com/@expo/config-plugins"
   target="_blank">
-  <img
-    alt="Downloads"
+  <ImageSpotlight
+    alt="Expo Config Plugins downloads."
     src="https://img.shields.io/npm/dm/@expo/config-plugins.svg?style=flat-square&labelColor=gray&color=33CC12&label=Downloads"
+    style={{ maxWidth: 480 }}
```
Comment (Simek) [inline docs/pages/workflow/continuous-native-generation.mdx:45]: I still think that this asset is too big, and in general, I don't like the idea of using shield in page content.

Screenshot 2023-06-19 at 10 39 22

**Review by Simek — APPROVED** [review]
Thank you for making those corrections and aligning content to our patterns! 🙏

File: docs/pages/workflow/continuous-native-generation.mdx (hunk 1 above)
Comment (amandeepmittal) [reply docs/pages/workflow/continuous-native-generation.mdx:45]: You're right. Making the maxWidth to `200` makes it a bit more presentable:

CleanShot 2023-06-19 at 14 13 01@2x


What do you suggest as a replacement? A text link?

Comment (Simek) [reply docs/pages/workflow/continuous-native-generation.mdx:45]: Doesn't have a good replacement idea, thus did not mark it as blocker for the PR. Shrinking should be sufficient for now, we can think about alternate solution in the future.

Comment (amandeepmittal) [reply docs/pages/workflow/continuous-native-generation.mdx:45]: Thank you!

**Pull request merged**